### PR TITLE
Disable sort optimization in search_after and scroll requests

### PR DESCRIPTION
**Pull request by dnhatn**
We can't enable the sort optimization with points in scroll requests until [LUCENE-10119](https://issues.apache.org/jira/browse/LUCENE-10119) is integrated. 

I labelled this PR non-issue for an unreleased bug.

### CODE REVIEW

**Review by jimczi — COMMENTED** [review]
Should we disable for all request with `search_after` and not only scroll ones ?

**Comment by dnhatn** [comment]
@jimczi I think search_after should be okay, but it's safer to disable them all now.

**Comment by dnhatn** [comment]
@jimczi I've disabled the optimization for search_after requests in https://github.com/elastic/elasticsearch/pull/78230/commits/6d8ea42df180dbff6dff5d41526923260d0ceebc. Can you take another look? Thank you!

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java [hunk 1]
```diff
@@ -136,6 +137,14 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
             assert query == searcher.rewrite(query); // already rewritten
 
             final ScrollContext scrollContext = searchContext.scrollContext();
+            if (searchContext.sort() != null && searchContext.sort().sort.getSort().length == 1) {
+                // We must disable the sort optimization with point for search_after or scroll requests until LUCENE-10119 is integrated.
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/search/query/QueryPhase.java:141]: Can you add a TODO to make it clear that an action is needed

**Review by jimczi — APPROVED** [review]
LGTM, thanks @dnhatn

**Comment by mayya-sharipova** [comment]
Elasticsearch `search_after` is not affected by this, and I think we don't need to disable it.

**Comment by dnhatn** [comment]
@mayya-sharipova I agree with your analysis. @jimczi Do you still prefer to disable the sort optimization in search_after requests?

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/search/query/QueryPhase.java]: added in https://github.com/elastic/elasticsearch/pull/78230/commits/4cdc73a6b6003fbf7277cee239b1f35b58839c44

**Comment by jimczi** [comment]
> or if a user is using _doc or _shard_doc with the main sort, there will not be a single sort

Not necessarily, `_shard_doc` can be used as the single sort in a PIT.

It's not really an issue anyway since we don't intend to release any version with this regression.

**Comment by mayya-sharipova** [comment]
> Not necessarily, _shard_doc can be used as the single sort in a PIT.

I think sort optimization is not applied for long fields, and is not applicable if there is a single sort `_shard_doc` and it does't use NumericCompararor.

But I am also ok to disable sort optimization with search_after completely for now to be completely sure we don't access it wrongly interally.

**Review by mayya-sharipova — APPROVED** [review]
@dnhatn Thanks a lot.

**Comment by dnhatn** [comment]
run elasticsearch-ci/part-1

**Comment by dnhatn** [comment]
@jimczi @mayya-sharipova Thanks for reviews.

**Pull request merged**